airbyte-commons/src/main/java/io/airbyte/commons/logging/MaskedDataInterceptor.java

```diff
@@ -0,0 +1,138 @@
+ * project. The file is named {@code specs_secrets_mask.yaml} and is located in the
+ * {@code src/main/resources/seed} directory of the {@link :airbyte-config:init} project.
+ */
+@Plugin(name = "MaskedDataInterceptor",
+        category = "Core",
+        elementType = "rewritePolicy",
+        printObject = true)
+public class MaskedDataInterceptor implements RewritePolicy {
+
+  protected static final Logger logger = StatusLogger.getLogger();
+
+  /**
+   * Mask string used as a replacement value for any sensitive properties found in the log message.
+   */
+  public static final String MASK = "********";
+
+  /**
+   * The pattern used to determine if a message contains sensitive data.
+   */
+  private final Optional<String> pattern;
+
+  @PluginFactory
+  public static MaskedDataInterceptor createPolicy(
+                                                   @PluginAttribute(value = "specMaskFile",
+                                                                    defaultString = "/seed/specs_secrets_mask.yaml") final String specMaskFile) {
+    return new MaskedDataInterceptor(specMaskFile);
+  }
+
+  private MaskedDataInterceptor(final String specMaskFile) {
+    this.pattern = buildPattern(specMaskFile);
+  }
+
+  @Override
+  public LogEvent rewrite(final LogEvent source) {
+    return Log4jLogEvent.newBuilder()
+        .setLoggerName(source.getLoggerName())
+        .setMarker(source.getMarker())
+        .setLoggerFqcn(source.getLoggerFqcn())
+        .setLevel(source.getLevel())
+        .setMessage(new SimpleMessage(applyMask(source.getMessage().getFormattedMessage())))
+        .setThrown(source.getThrown())
+        .setContextMap(source.getContextMap())
+        .setContextStack(source.getContextStack())
+        .setThreadName(source.getThreadName())
+        .setSource(source.getSource())
+        .setTimeMillis(source.getTimeMillis())
+        .build();
+  }
+
+  /**
+   * Applies the mask to the message, if necessary.
+   *
+   * @param message The log message.
+   * @return The possibly masked log message.
+   */
+  private String applyMask(final String message) {
+    if (pattern.isPresent()) {
+      return message.replaceAll(pattern.get(), "\"$1\":\"" + MASK + "\"");
+    } else {
+      return message;
+    }
+  }
+
+  /**
+   * Loads the maskable properties from the provided file.
+   *
+   * @param specMaskFile The spec mask file.
+   * @return The set of maskable properties.
+   */
+  private Set<String> getMaskableProperties(final String specMaskFile) {
+    logger.info("Loading mask data from '{}", specMaskFile);
+    try {
+      final String maskFileContents = IOUtils.toString(getClass().getResourceAsStream(specMaskFile), Charset.defaultCharset());
+      final Map<String, Set<String>> properties = Jsons.object(Yamls.deserialize(maskFileContents), new TypeReference<>() {});
+      return properties.getOrDefault("properties", Set.of());
```

Comment:
So this will mask everything by default in the log even the fields that we not tagged as `airbyte_secret: true`, Right?

Reply:
@benmoriceau No.  The mask file only contains the name of properties in the spec that have `airbyte_secret: true` and the generated regular expression only looks for JSON snippets in the log message that contains one of those property names.  If we cannot load the file, then it doesn't attempt any replacement.

It's worth noting that as it is currently implemented in this PR, the replace will not find other variations, even for similar strings.  For instance, "This is a log message with password: foo" will not be found even if `password` is one of the fields.  The string would have to look like JSON (e.g. `"password":"foo"`) for the replace to occur.  Obviously, this is just a choice and we can add/extend this re-write logic as we go to catch other pathological cases (e.g. all caps ENV VARS, etc).

Reply:
> If we cannot load the file, then it doesn't attempt any replacement.
Should we fail hard instead?

Reply:
@gosusnp I'm not sure that is possible without some additional code to check that the file exists from the actual application code.  In testing, before I added some fixes, you just end up with some log4j errors when the file is not found and that does not prevent the application from running -- it just creates a bunch of logging noise.  If we decide that we really want to prevent startup if we cannot mask, we would also need to add a utility class to check if the file is on the classpath in order to stop the service if not present.